spec/20-BINARY_FORMAT.md

```diff
@@ -1,3 +1,125 @@
+   the first byte.
+2. Read the `version` byte at the cursor position. Shift cursor to `1` byte.
+3. If all three fields (`trace-id`, `parent-id`, `trace-flags`) already read -
+   return them with the status `OK` if `version` is `0` or status
+   `DOWNGRADED_TO_ZERO` otherwise.
+4. If at the end of the buffer return invalid status `TRACEPARENT_INCOMPLETE`.
+   Otherwise read the field identifier byte at the cursor position. Field
+   identifier should be read as unsigned byte assuming big-endian bits order.
+    1. If `0` - check that remaining buffer size is more or equal to `16` bytes.
+       If shorter - return invalid status `TRACE_ID_TOO_SHORT`. Otherwise read
+       the next `16` bytes for `trace-id` and shift cursor to the end of those
+       `16` bytes. Go to step `3`. If `trace-id` is represented as a byte array
+       - first byte should be set into the first element of that array. See
+         comment in serialization section.
+    2. If `1` - check that remaining buffer size is more or equal to `8` bytes.
+       If shorter - return invalid status `PARENT_ID_TOO_SHORT`. Otherwise read
+       the next `8` bytes for `parent-id` and shift cursor to the end of those
+       `8` bytes. Go to step `3`.
+    3. If `2` - check the remaining size of the buffer. If at the end of the
+       buffer - return invalid status. Otherwise - read the `trace-flags`
+       byte. Least significant bit will represent `recorded` value. Go to step
+       `3`.
+    4. In case of any other value - if `version` read at step `2` is `0` -
+       return invalid status `INVALID_FIELD_ID`. If `version` has any other
+       value - `INCOMPATIBLE_VERSION`.
+
+_Note_, that invalid status names are given for readability and not part of the
+specification.
+
+_Note_, that parsing should not treat any additional bytes in the end of the
+buffer as an invalid status. Those fields can be added for padding purposes.
+Optionally implementation can check that the buffer is longer than `29` bytes as
+a very first step if this check is not expensive.
+
+## `traceparent` example
+
+``` js
+{0,
+  0, 75, 249, 47, 53, 119, 179, 77, 166, 163, 206, 146, 157, 0, 14, 71, 54,
+  1, 52, 240, 103, 170, 11, 169, 2, 183,
+  2, 1}
+```
+
+This corresponds to:
+
+- `trace-id` is
+  `{75, 249, 47, 53, 119, 179, 77, 166, 163, 206, 146, 157, 0, 14, 71, 54}` or
+  `4bf9273577b34da6a3ce929d000e4736`.
+- `span-id` is `{52, 240, 103, 170, 11, 169, 2, 183}` or `34f067aa0ba902b7`.
+- `trace-flags` is `1` with the meaning `recorded` is true.
+
+## `tracestate` binary format
+
+List of up to 32 name-value pairs. Each list member starts with the 1 byte field
+identifier `0`. The format of list member is a single byte key length followed
+by the key value and single byte value length followed by the encoded value.
+Strings are transmitted in ASCII encoding.
+
+``` abnf
+tracestate      = list-member 0*31( list-member )
+list-member     = "0" key-len key value-len value
```

Comment:
[Comment](https://github.com/w3c/trace-context/pull/215/files#r239377574) was resolved in original PR:

![image](https://user-images.githubusercontent.com/9950081/53045740-38171200-3443-11e9-8b6c-26c5545f58a8.png)